### ISSUE TITLE
Investigate issues with FlexBox and other CSS Flexible layouts

### ISSUE THREAD

**Issue by awkawk** [w3c/wcag#83]
Is anyone hearing about issues with this from web content in production?

**Comment by awkawk** [comment]
James says that the issues he sees are most often related to mismatch between visual and focus order as a result of form controls and links within a grid layout.  People need to be careful in development.

**Comment by marcjohlic** [comment]
Some notes from when I was looking into flexbox:
* Flexbox allows page content to be reordered 
* Potential Impacts to WCAG SC 1.3.2 Meaningful Seq and 2.4.3 Focus Order ?
* [5.4.1 "Reordering and Accessibility"] (http://www.w3.org/TR/css3-flexbox/#order-accessibility) in the flexbox spec should cover WCAG 1.3.2, but we would have to put together the Technique(s) (and/or Failures) for WCAG.  
  * Something around the proper use of the "order" property - logical vs visual order
  * Could probably write something based on the example they have in 5.4.1
  * Have other working groups (PF, CSS?) come to a consensus on whether flexbox 'order' should affect AT speech output?
* WCAG 2.4.3 Focus Order - do we need a Technique or Failure that would talk to changes being introduced to the Tab order
  * If navigation affects meaning of the page focusable components receive focus in that same order
  * Impacts to techniques using tabindex (nav-index)
* [Flexbox accessibility] (https://wiki.csswg.org/spec/css3-flexbox/accessibility) from CSS Working Group wiki for reference

**Comment by joshueoconnor** [comment]
@marcjohlic  That looks very interesting, it would be good to fold this info into any related techiques.

**Comment by joshueoconnor** [comment]
Oops.. didn't mean to close. Its reopened.

**Comment by LjWatson** [comment]
Didn't realise there was an open issue on this. Blogged about flexbox and keyboard navigation yesterday:
http://tink.uk/flexbox-the-keyboard-navigation-disconnect/

**Comment by joshueoconnor** [comment]
Excellent @LjWatson It would be cool if you teamed up with @marcjohlic  on this *grin.

**Comment by marcjohlic** [comment]
Suggest this be marked as Deferred  and closed (per the [Issues process] (https://www.w3.org/WAI/GL/wiki/Handling_Issues)) until such time as there are CSS / Flexbox SME resources available in the group that can investigate this issue further.

**Comment by inclusiveThinking** [comment]
I am new to the WCAG 2.0 Working Group and ready to contribute. Would be open to doing in depth research on this if it is important for the group.

**Comment by michael-n-cooper** [comment]
The APA Working Group is working with the CSS WG on issues with Flexbox specifically and CSS layouts in general. Specific tracking on Flexbox is at https://www.w3.org/WAI/APA/wiki/CSS_Flexible_Box_Layout_Module_Level_1 but other work is starting up. I recommend the WCAG WG defer this issue until those processes complete. Anybody interested in participating within APA could join the work there. https://www.w3.org/WAI/APA/participation

**Comment by awkawk** [comment]
Preparing to mark as deferred due to inactivity and group resource constraints (https://www.w3.org/WAI/GL/wiki/Handling_Issues).

**Comment by benmcginnis** [comment]
Hi all, when working to be WCAG compliant, should authors consider "[Authors must use order only for visual, not logical, reordering of content. Style sheets that use order to perform logical reordering are non-conforming](http://www.w3.org/TR/css3-flexbox/#order-accessibility)" to be part of the WCAG 2.0 spec?  

If so, what defines logical order? Is it only focus order, or does the document as spoken/linearized have to match the visual presentation to be conforming?  Would really appreciate some clarification in [understanding 1.3.2](/w3c/wcag/blob/master/wcag20/sources/understanding/content-structure-separation-programmatic.xml)

**Comment by DavidMacDonald** [comment]
Basically it means, can the user, use the page.
